src/sensors/SensorManager.cpp

```diff
@@ -33,109 +33,171 @@
 #include "ErroneousSensor.h"
 #include "sensoraddresses.h"
 
+#if ESP32
+    #include "driver/i2c.h"
+#endif
+
 namespace SlimeVR
 {
     namespace Sensors
     {
-        void SensorManager::setup()
+        Sensor* SensorManager::buildSensor(uint8_t sensorID, uint8_t imuType, uint8_t address, float rotation, uint8_t sclPin, uint8_t sdaPin, int extraParam)
         {
-            uint8_t firstIMUAddress = 0;
-            uint8_t secondIMUAddress = 0;
+            m_Logger.trace("Building IMU with: id=%d,\n\
+                            imuType=0x%02X, address=0x%02X, rotation=%f,\n\
+                            sclPin=%d, sdaPin=%d, extraParam=%d",
+                            sensorID,
+                            imuType, address, rotation,
+                            sclPin, sdaPin, extraParam);
+
+            // Now start detecting and building the IMU
+            Sensor* sensor = NULL;
+            
+            // Clear and reset I2C bus for each sensor upon startup
+            I2CSCAN::clearBus(sdaPin, sclPin);
+            swapI2C(sclPin, sdaPin);
+
+            if (I2CSCAN::isI2CExist(address)) {
+                m_Logger.trace("IMU %d found at address 0x%02X", sensorID, address);
+            } else {
+                sensor = new ErroneousSensor(sensorID, imuType);
```

Comment:
Couldn't we just return the sensor here?

Reply:
That is for reproducing the previous behavior in L54-L57 here:
https://github.com/SlimeVR/SlimeVR-Tracker-ESP/blob/48d87a327eb8bc1bd4d4174a3aa549c18011937f/src/sensors/SensorManager.cpp#L54-L57

Reply:
Oh, I just realized, you're never calling `motionSetup()` on this sensor.